composer/loss/utils.py

```diff
@@ -47,23 +46,37 @@ def ensure_targets_one_hot(input: torch.Tensor,
         num_classes (int, optional): Number of classes. If not specified, this will be inferred
             from input. Default: ``None``
     """
+
     if infer_target_type(input, targets) == 'indices':
         # If the number of classes isn't specified, attempt to infer it from the input
         if num_classes is None:
             num_classes = input.shape[1]
         if targets.min() < 0:
             warnings.warn('Negative label indices are being ignored in conversion to one-hot labels')
+
+            # Add new dimension for the class vectors
+            class_dim = 1
+            targets = targets.clone().unsqueeze(class_dim).long()
+
             # Map all negative indicies to a class to drop.
             targets[targets < 0] = num_classes
-            targets = F.one_hot(targets, num_classes=num_classes + 1)
```

Comment:
Prefer to use the torch one_hot, I'm not sure a 10% optimization on the loss calculation (on your benchmarked tensor sizes) is worth the implementation complexity? or at minimum, refactor the MONAI-based implementation into a `def _one_hot(..)` method for readability?

otherwise, it's unclear here what changes here are refactoring the one hot comptuation, and which are net new functional changes.

Reply:
The throughput without dice is ~210 images/sec. With the dice implementation that uses `F.one_hot`, it is ~170 images/sec. With the dice implementation with the new one hot conversion, it is ~190 images/sec. To me, this seems like a significant difference in throughput to justify the implementation change . Similarly, we see significant throughput differences with mixup and label smoothing, but the new implementation should see less impact. 

I can move the implementation to separate function.

Reply:
OK didn't realize the 10% was in overall throughput, not just the loss calculation itself. Then yes that's significant -- can you move to a separate function, thanks!

Reply:
I ended up moving most of the `ensure_targets_one_hot` functionality to the `_one_hot` function since it seemed to fit. Let me know if it matches expectations

Reply:
Ideally `_one_hot` would be a drop-in replacement for `F.one_hot` that is faster, and respects the same semantics.  That also makes it easy to add tests, as there exists a reference implementation.

Is that possible in this case (e.g. keep the same code, but just replace `F.one_hot` -> `_one_hot`?), or is the optimization mixed in with some of the other bushiness logic?